lib/active_merchant/billing/gateways/redsys_rest.rb

```diff
@@ -361,7 +361,12 @@ def commit(post, options)
         payload = raw_response['Ds_MerchantParameters']
         return Response.new(false, "#{raw_response['errorCode']} ERROR") unless payload
 
-        response = JSON.parse(Base64.decode64(payload)).transform_keys!(&:downcase).with_indifferent_access
+        begin
+          response = JSON.parse(Base64.decode64(payload)).transform_keys!(&:downcase).with_indifferent_access
```

Comment:
@DustinHaefele does the Base64 interface really ask that you catch an exception and then try another method in the interface? if you don't mind tell me a little more about it, although I know there is urgency. the thing that kinda bothers me is--what if `urlsafe_decode64` raises an error.

Reply:
Great question.

The issue is with a standard base64 encoding as opposed to the urlsafe_encoding there are different character sets. In this case when we are trying to do a standard base64 decoding, if there is a `?` it will not decode correctly and it will break the JSON syntax because everything after the `?` will be something like. `\x0FY\x0ELM\x18\x8DL\xD8\xCBML\x8DL\xCBM\` and wont end up with the correct closing brackets. 

It's likely that we could always use the `urlsafe_decode64` and remove the first block but with an emergency change I wanted to be abundantly safe and only fix the specific error that we are seeing which we know the root cause of...That damn `?`... I didn't do it whole sale because I'm not sure if there are characters that will cause issues while using the urlsafe_decode, in the same way that we are seeing the `?` cause issues here. I would like more time to explore that and then put the change in whole sale with a ticket to address it. 

If the urlsafe_decode throws an error after the standard decode throws an error than it is a different issue, and I'm fine with an exception being raised.

Reply:
ok cool I will recuse myself. makes sense